src/tests/subcommands/add_test.go

```diff
@@ -0,0 +1,104 @@
+package tests
+
+import (
+	"bytes"
+	"encoding/gob"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"accretional.com/semantifly/subcommands"
+)
+
+func createTempFile(t *testing.T, dir string, data []byte) *os.File {
+	file, err := os.CreateTemp(dir, "testfile")
+	if err != nil {
+		t.Fatalf("Failed to create temp file: %v", err)
+	}
+	_, err = file.Write(data)
+	if err != nil {
+		t.Fatalf("Failed to write to temp file: %v", err)
+	}
+	err = file.Close()
+	if err != nil {
+		t.Fatalf("Failed to close temp file: %v", err)
+	}
+	return file
+}
+
+func createTempDir(t *testing.T) string {
```

Comment:
ditto here, shouldn't we remove t? Also, I think you should either have this function take a parameter of the filepath or simply remove the function

Reply:
Thought I might need to reuse the function for future test cases, but not really needed. Took it off